crates/primitives/src/transaction/error.rs

```diff
@@ -52,3 +52,14 @@ pub enum InvalidTransactionError {
     #[error("transaction signer has bytecode set")]
     SignerAccountHasBytecode,
 }
+
+/// Represents error variants that can happen when trying to convert a transaction to
+/// [`PooledTransactionsElement`](crate::PooledTransactionsElement)
+#[derive(Debug, Clone, Eq, PartialEq, thiserror::Error)]
+pub enum TransactionConversionError {
+    /// This error variant is used when a transaction cannot be converted into a
+    /// [`PooledTransactionsElement`](crate::PooledTransactionsElement) because it is not supported
+    /// for broadcasting over the P2P network.
```

Comment:
this should not mention broadcast because there are two types of p2p messages where txs are exchanged:
broadcast, request-response